dagshub/upload/wrapper.py

```diff
@@ -198,7 +197,15 @@ def upload(
 
         """
         if os.path.isdir(local_path):
-            dir_to_upload = self.directory(remote_path if remote_path is not None else DEFAULT_REMOTE_PATH)
+            if remote_path is None:
+                abspath = os.path.abspath(local_path)
+                cwd = os.getcwd()
+                if abspath.startswith(cwd):
+                    remote_path = os.path.relpath(abspath, cwd)
+                else:
+                    remote_path = os.path.basename(abspath)
+            remote_path = remote_path.replace(os.sep, '/')
+            dir_to_upload = self.directory(remote_path)
```

Comment:
Check out `pathlib` sometime, because some of the stuff that it can do with paths is so much nicer.
https://docs.python.org/3.7/library/pathlib.html
(Mind the 3.7, they get a lot of neat features later, but 3.7 is our baseline)

I think this code can do the same thing as what you did here, but instead of string mangling and using os libs it's all object manipulations
```python
lp = Path(local_path).resolve()
cwd = Path(".").resolve()
try:
    relative_path = lp.relative_to(cwd)
except ValueError:
    relative_path = Path(lp.name)
remote_path = relative_path.as_posix()
dir_to_upload = self.directory(remote_path)
```

Reply:
I tried but so far it's having edge cases.
Namely, if `local_path` is just a non-absolute path, then `local_path.relative_to(Path.cwd())` fails with `ValueError` and ruins everything.
I guess because it requires using `join` but that makes everything way more complicated